donation-tracker/src/utils/dataStore.ts

```diff
@@ -1,316 +1,486 @@
+        return false;
+      }
+    }
+    return true;
+  }
 
-	/**
-	 * Function to check if the local data is outdated (older then the configured amount of minutes).
-	 * @returns A boolean result with the answer to the question
-	 */
-	private isDataOutdated() {
-		this.debugLog('DataStore: Checking if data is outdated...');
+  /**
+   * Internal function to end class init with no local data
+   * @param logMessage Message that should be logged as reason
+   */
+  private initWithEmptyLocalData(logMessage: string) {
+    this.debugLog(logMessage);
+    this.localData = null;
+    this.hasLocalData = false;
+    if (!this.isGatsbyBuild) setTimeout(this.refreshData.bind(this), 100);
+  }
+
+  /**
+   * Internal function to register the next data update
+   */
+  private registerDataUpdate() {
+    if (PageConfiguration.AutoRefresh) {
+      let secondsUntilRefresh = this.getSecondsUntilRefresh();
+      if (secondsUntilRefresh == undefined) {
+        // use default refresh time if it can not be calculated
+        secondsUntilRefresh = 1000 * 60 * this.maxDataAgeInMinutes;
+      }
+      this.debugLog(`DataStore: Registering next data refresh to execute in ${secondsUntilRefresh} seconds`);
+      setTimeout(this.refreshData.bind(this), secondsUntilRefresh * 1000);
+    }
+  }
+
+  /**
+   * Internal function to map the columns to the data
+   * @param csvConfigRow The first row of the CVS file / export
+   * @returns A ColumnMapping object
+   */
+  private createColumnMapping(csvConfigRow: string): ColumnMapping {
+    this.debugLog('DataStore: Creating column mapping');
+    const headers = this.parseCsvLine(csvConfigRow.toLowerCase());
+    const result: ColumnMapping = {
+      languages: new Array<string>(),
+      item: {},
+      unit: {},
+      form: {},
+      campaignId: -1,
+      need: -1,
+      donated: -1,
+      lastChange: -1,
+    };
+    headers.forEach((columnContent, index) => {
+      if (columnContent.toString().indexOf(':') > -1) {
+        // we have a localized value here
+        const pair = columnContent.toString().split(':');
+        const langCode = pair[1].toString().trim();
+        let langIndex = result.languages.indexOf(langCode);
+        if (langIndex == -1) {
+          result.languages.push(langCode);
+        }
+        switch (pair[0].trim()) {
+          case 'item':
+            result.item[langCode] = index;
+          case 'unit':
+            result.unit[langCode] = index;
+          case 'form':
+            result.form[langCode] = index;
+        }
+      } else {
+        // we have a non localized value
+        switch (columnContent.toString().trim()) {
+          case 'campaign':
+            result.campaignId = index;
+          case 'need':
+            result.need = index;
+          case 'donated':
+            result.donated = index;
+          case 'lastchange':
+            result.lastChange = index;
+        }
+      }
+    });
+    return result;
```

Comment:
I wonder, if we can not reshape the whole function in the following way (rough sketch):
```
const headers = this.parseCsvLine(csvConfigRow.toLowerCase()).map(columnContent => columnContent.toString().trim());
return = {
      languages: uniq(headers.filter(header => header.includes(':'))) // or similar to maybe not use lodash
      item: headers.filter(header => header.includes('item:')).reduce((result, header, index)=>({...result, [getLangCode(header)]: index}), {})
      unit: ...,
      form: ...,
      campaignId: headers.indexOf(header => header === 'campaign'),
      need: headers.indexOf(header => header === 'need'),
      donated: headers.indexOf(header => header === 'donated'),
      lastChange: headers.indexOf(header => header === 'lastchange'),
    };
```